src/bd/amazed/pdfscissors/model/TaskPdfOpen.java

```diff
@@ -54,27 +50,19 @@ public TaskPdfOpen(File file, int groupType, boolean shouldCreateStackView, Comp
 
 			if (shouldCreateStackView && groupType != PageGroup.GROUP_TYPE_INDIVIDUAL) {
 				setProgress(0);
-				PropertyChangeListener propertyChangeListener = new PropertyChangeListener() {
-					@Override
-					public void propertyChange(PropertyChangeEvent evt) {
-						firePropertyChange(evt.getPropertyName(), evt.getOldValue(), evt.getNewValue());
-					}
-				};
-
-				for (int i = 0; i < pageGroups.size(); i++) {
-					PageGroup pageGroup = pageGroups.elementAt(i);
-					BufferedImage image = cropper.getImage(propertyChangeListener, pageGroup);
-
-					if (image == null) {
-						debug("Ups.. null image for " + pdfFile.getNormalizedFile());
-					} else {
-						debug("PDF loaded " + pageGroup + " from " + pdfFile.getNormalizedFile());
-					}
-					pageGroup.setStackImage(image);
-
-				}
+				TaskCreateThumbnails thumbnailTask = launchThumbnailTask(pdfFile, pageGroups);
+				
+				//make this thread wait, until the thumbnails-thread has finished:
+				// (images are previews for stacked-pages, i.e. required)
+				thumbnailTask.get();
+				
 				setProgress(100);
 			}
+			else {
+				//start thumbnails-thread without waiting for it to finish
+				// (images are only previews for single pages, i.e. not required)
+				launchThumbnailTask(pdfFile, pageGroups);

```

Comment:
The checkbox 'Create stack view' that is under advanced option was created because some users had complained about handling of extra large files. If user explicitly declare 'dont create stack view' perhaps we dont need to create the stack view.